### PR TITLE
ispc: unstable-2021-04-02 -> 1.18.0, (co-)maintain

### PR DESCRIPTION
**Pull request by athas**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by athas** [comment]
@aristid @thoughtpolice could you take a look at this one?

**Comment by Mindavi** [comment]
@ofborg eval

**Comment by athas** [comment]
I have taken the liberty of adding myself as co-maintainer, as the current maintainers seem to be absent.

File: pkgs/development/compilers/ispc/default.nix [hunk 1]
```diff
@@ -1,41 +1,29 @@
...
-    (fetchpatch {
-      url = "https://github.com/erictapen/ispc/commit/338119b2f4e11fcf0b0852de296c320928e572a2.patch";
-      sha256 = "sha256-+RqDq1LMWomu/K4SgK0Nip47b1RwyM6W0cTSNGD4+m4=";
-    })
-  ];
-
   nativeBuildInputs = [ cmake which m4 bison flex python3 llvmPackages.llvm.dev ];
   buildInputs = with llvmPackages; [
-    llvm llvmPackages.libclang
+    llvm libclang openmp ncurses
```
Comment (sternenseemann) [inline pkgs/development/compilers/ispc/default.nix:21]: Technically the same, but a bit clearer.

```suggestion
    libllvm libclang openmp ncurses
```

File: pkgs/development/compilers/ispc/default.nix [hunk 2]
```diff
@@ -1,41 +1,29 @@
...
-  patches = [
-    # Fix cmake error: `Failed to find clang++`
-    # https://github.com/ispc/ispc/pull/2055
-    (fetchpatch {
-      url = "https://github.com/erictapen/ispc/commit/338119b2f4e11fcf0b0852de296c320928e572a2.patch";
-      sha256 = "sha256-+RqDq1LMWomu/K4SgK0Nip47b1RwyM6W0cTSNGD4+m4=";
-    })
-  ];
-
   nativeBuildInputs = [ cmake which m4 bison flex python3 llvmPackages.llvm.dev ];
```
Comment (sternenseemann) [inline pkgs/development/compilers/ispc/default.nix:19]: `libllvm.dev` makes it clearer that this is for `llvm-config` / `llvm-config-native` the exception where we wouldn't want to take LLVM from `buildPackages`. Adding a comment may also not hurt…

```suggestion
  nativeBuildInputs = [ cmake which m4 bison flex python3 llvmPackages.libllvm.dev ];
```

**Review by sternenseemann — COMMENTED** [review]
Some LLVM related nitpicks, other than that it looks good.

The Release Notes say something about LLVM 13, currently it'd be built with LLVM 11 AFAICT, is that supported?

**Comment by sternenseemann** [comment]
@ofborg build openimagedenoise embree openimagedenoise_1_2_x

File: pkgs/development/compilers/ispc/default.nix (hunk 1 above)
Comment (athas) [reply pkgs/development/compilers/ispc/default.nix]: Thanks; fixed.

**Pull request merged**